crates/cadenza-gcode/build/test_data.rs

```diff
@@ -0,0 +1,68 @@
+pub fn tests() -> String {
+    let examples = Example::load("test-data");
+    let mut out = String::new();
+    macro_rules! w {
+        ($($tt:tt)*) => {
+            out.push_str(&format!($($tt)*));
+            out.push('\n');
+        };
+    }
+
+    w!("use crate::parse;");
+    w!("use insta::assert_snapshot as s;");
+    w!("");
+
+    // Generate parse and AST snapshot tests for each example
+    for Example { name, src } in examples.iter() {
+        w!("mod {name} {{");
+        w!("    use super::*;");
+        w!("    #[test]");
+        w!("    fn parse_ast() {{");
```

Comment:
We also need a test to show the CST as well. My guess is we're still not attributing all bytes in the file to a token type. Like the whitespace and newlines definitely aren't. Neither is the comment start. Showing the CST would show that we have missing characters.